api/api-plugin-infos-v5/src/main/java/com/thoughtworks/go/apiv5/plugininfos/representers/extensions/ConfigRepoExtensionRepresenter.java

```diff
@@ -16,5 +16,22 @@
 
 package com.thoughtworks.go.apiv5.plugininfos.representers.extensions;
 
+import com.thoughtworks.go.api.base.OutputWriter;
+import com.thoughtworks.go.plugin.domain.common.PluginInfo;
+import com.thoughtworks.go.plugin.domain.configrepo.ConfigRepoPluginInfo;
+
 public class ConfigRepoExtensionRepresenter extends ExtensionRepresenter {
+    @Override
+    public void toJSON(OutputWriter extensionWriter, PluginInfo extension) {
+        super.toJSON(extensionWriter, extension);
+
+        ConfigRepoPluginInfo configRepoPluginInfo = (ConfigRepoPluginInfo) extension;
+
+        if (configRepoPluginInfo.getCapabilities() != null) {
```

Comment:
No need to add the `null` check. We always ask the plugin to send capabilities. In absence of capabilities, it should fail much earlier while building plugin info.

Reply:
Ok that makes sense. I'll remove it.